pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/logicaltable/BaseLogicalTableIntegrationTest.java

```diff
@@ -0,0 +1,515 @@
+  protected static final String DEFAULT_TABLE_NAME = "physicalTable";
+  private static final int NUM_OFFLINE_SEGMENTS = 12;
+  protected static BaseLogicalTableIntegrationTest _sharedClusterTestSuite = null;
+
+  @BeforeSuite
+  public void setUpSuite()
+      throws Exception {
+    LOGGER.info("Setting up integration test suite");
+    TestUtils.ensureDirectoriesExistAndEmpty(_tempDir, _segmentDir, _tarDir);
+    _sharedClusterTestSuite = this;
+
+    // Start the Pinot cluster
+    startZk();
+    LOGGER.info("Start Kafka in the integration test suite");
+    startKafka();
+    startController();
+    startBroker();
+    startServers(2);
+    LOGGER.info("Finished setting up integration test suite");
+  }
+
+  @AfterSuite
+  public void tearDownSuite()
+      throws Exception {
+    LOGGER.info("Tearing down integration test suite");
+    // Stop Kafka
+    LOGGER.info("Stop Kafka in the integration test suite");
+    stopKafka();
+    // Shutdown the Pinot cluster
+    stopServer();
+    stopBroker();
+    stopController();
+    stopZk();
+    FileUtils.deleteDirectory(_tempDir);
+    LOGGER.info("Finished tearing down integration test suite");
+  }
+
+  @Override
+  protected String getTableName() {
+    return DEFAULT_TABLE_NAME;
+  }
+
+  @BeforeClass
+  public void setUp()
+      throws Exception {
+    TestUtils.ensureDirectoriesExistAndEmpty(_tempDir, _segmentDir, _tarDir);
+    if (_sharedClusterTestSuite != this) {
+      _controllerRequestURLBuilder = _sharedClusterTestSuite._controllerRequestURLBuilder;
+    }
+
+    List<File> avroFiles = getAllAvroFiles();
+    int numSegmentsPerTable = NUM_OFFLINE_SEGMENTS / getOfflineTableNames().size();
+    int index = 0;
+    for (String tableName : getOfflineTableNames()) {
+      File tarDir = new File(_tarDir, tableName);
+
+      TestUtils.ensureDirectoriesExistAndEmpty(tarDir);
+
+      // Create and upload the schema and table config
+      Schema schema = createSchema(getSchemaFileName());
+      schema.setSchemaName(tableName);
+      addSchema(schema);
+      TableConfig offlineTableConfig = createOfflineTableConfig(tableName);
+      addTableConfig(offlineTableConfig);
+
+      List<File> offlineAvroFiles = new ArrayList<>(numSegmentsPerTable);
+      for (int i = index; i < index + numSegmentsPerTable; i++) {
+        offlineAvroFiles.add(avroFiles.get(i));
+      }
+      index += numSegmentsPerTable;
+
+      // Create and upload segments
+      ClusterIntegrationTestUtils.buildSegmentsFromAvro(offlineAvroFiles, offlineTableConfig, schema, 0, _segmentDir,
+          tarDir);
+      uploadSegments(tableName, tarDir);
+    }
+
+    createLogicalTable();
+
+    // Set up the H2 connection
+    setUpH2Connection(avroFiles);
+
+    // Initialize the query generator
+    setUpQueryGenerator(avroFiles);
+
+    // Wait for all documents loaded
+    waitForAllDocsLoaded(600_000L);
+  }
+
+  @AfterClass
+  public void tearDown()
```

Comment:
don't we need to cleanup the H2 setup?

Reply:
Interesting. I dont see any cleanup at all. Also I tried to set it up once in `BeginSuite` and that didnt work because of temporary file setup etc.